segments_tree.go

```diff
@@ -0,0 +1,141 @@
+package quamina
+
+import (
+	"fmt"
+	"strings"
+)
+
+const SegmentSeparator = "\n"
+
+type segmentsTree struct {
```

Comment:
So far I haven't figured out how this works at all.  I think for a type like this it's useful and conventional to have a comment block explaining the fields, as most of the other significant Quamina types do.

Reply:
I wrote such a comment on the interface it self `segments_tree_tracker.go`, adding a comment on segmentsTree will be the same comment from my side.

But I added some comments around the code - what is fields, nodes and for other methods with examples, hopefully it's clearer now.

Reply:
Ah, I had looked at the interface definition but hadn't really absorbed it, but now after looking at the code, this is making more sense. BTW, after we merge, I will probably do a PR to improve spelling and English language style (I hope that's OK) but it's not worth holding this up.